docs: Add major version upgrade doc, 1.x to 2.x upgrade (#311)
openedx/openedx-aspects: docs/technical_documentation/how-tos/upgrade.rst

event_sink  most_recent_taxonomies          event_sink  dim_most_recent_taxonomies
xapi        responses                       reporting   dim_learner_response_attempt
xapi        section_page_engagement         reporting   fact_section_page_engagement
xapi        section_problem_engagement      reporting   fact_section_problem_engagement
xapi        section_video_engagement        reporting   fact_section_video_engagement
xapi        subsection_page_engagement      reporting   fact_subsection_page_engagement
xapi        subsection_problem_engagement   reporting   fact_subsection_problem_engagement
xapi        subsection_video_engagement     reporting   fact_subsection_video_engagement
reporting   watched_video_duration          reporting   fact_watched_video_duration
==========  ==============================  ==========  ====================================

These model updates are done as part of the dbt step of init, or can be executed manually or even individually by running different forms of the dbt command. Moved or deprecated models will not have their old versions deleted as part of this upgrade, which should allow for a more seamless transition.

> Once the Superset assets are updated, the last step in the Aspects init process, the old models can and should be safely dropped. This can be done by running the following command:

Reviewer: ```suggestion
Once the Superset assets are updated (the last step in the Aspects init process), the old models can and should be safely dropped. This can be done by running the following command:
```

``tutor [dev|local|k8s] do dbt -c 'run-operation remove_deprecated_models' --only_changed False``

1.x to 2.x Upgrade Process
==========================

For most cases the upgrade process is simply:

- Update the version of the ``tutor-contrib-aspects`` plugin.

> - Optionally delete the ``/env/plugins/aspects/build/aspects-superset/`` from your Tutor directory to ensure that any old assets are removed. This depende on the version of the plugin you are upgrading from, but usually manifests as having the wrong language assets showing up in the Superset chargs and dashboards.

Reviewer: ```suggestion
- Optionally delete the ``/env/plugins/aspects/build/aspects-superset/`` from your Tutor directory to ensure that any old assets are removed. This depends on the version of the plugin you are upgrading from, but usually manifests as having the wrong language assets showing up in the Superset charts and dashboards.
```

- ``tutor config save``

> - ``tutor images build aspects aspects-superset --no-cache``

Reviewer: ```suggestion
- ``tutor images build aspects --no-cache aspects-superset --no-cache``
```
